lib/plugins/wallet/zcashd/zcashd.js

```diff
@@ -46,11 +46,42 @@ function balance (account, cryptoCode) {
 
 function sendCoins (account, address, cryptoAtoms, cryptoCode) {
   const coins = cryptoAtoms.shift(-unitScale).toFixed(8)
-
+  const checkSendStatus = function (opid) {
+    return new Promise((resolve, reject) => {
+      function checker(timeout) {
+        setTimeout(() => {
+          fetch('z_getoperationstatus', [[opid]])
+            .then(res => {
+              const status = _.get('status', res[0])
+              switch(status) {
+                case 'success': 
+                  resolve(res[0])
+                  break
+                case 'failed': 
+                  reject(res[0].error)
+                  break
+                case 'executing':
+                  if (timeout > 0) {
+                    checker(timeout - 500)
+                  } else {
+                    reject(res)
+                  }
+              }
+            })
+        }, 500)
+      }
+      checker(10000)
```

Comment:
Shouldn't this be ``return checker(10000)``?

Reply:
The `Promise` constructor expects some executor function that returns void, so we don't need to return anything from this function. The `checker` function returns whatever `setTimeout` returns, which I believe is an ID that identifies the timeout, so that it could be canceled somewhere else if needed, but this is not what we need.